cmd/determinize-prow-config/main.go

```diff
@@ -0,0 +1,83 @@
+// determinize-prow-config reads and writes Prow configuration
+// to enforce formatting on the files
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+
+	"github.com/sirupsen/logrus"
+	prowconfig "k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/plugins"
+	"sigs.k8s.io/yaml"
+)
+
+type options struct {
+	prowConfigDir string
+}
+
+func (o *options) Validate() error {
+	if o.prowConfigDir == "" {
+		return errors.New("--prow-config-dir is required")
+	}
+	return nil
+}
+
+func gatherOptions() options {
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	fs.StringVar(&o.prowConfigDir, "prow-config-dir", "", "Path to the Prow configuration directory.")
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		logrus.WithError(err).Fatal("could not parse input")
+	}
+	return o
+}
+
+func main() {
+	o := gatherOptions()
+	if err := o.Validate(); err != nil {
+		logrus.WithError(err).Fatal("invalid options")
+	}
+
+	if err := updateProwConfig(o.prowConfigDir); err != nil {
+		logrus.WithError(err).Fatal("could not update Prow configuration")
+	}
+
+	if err := updatePluginConfig(o.prowConfigDir); err != nil {
+		logrus.WithError(err).Fatal("could not update Prow plugin configuration")
+	}
+}
+
+func updateProwConfig(configDir string) error {
+	configPath := path.Join(configDir, "_config.yaml")
+	agent := prowconfig.Agent{}
+	if err := agent.Start(configPath, ""); err != nil {
+		return fmt.Errorf("could not load Prow configuration: %v", err)
+	}
+	data, err := yaml.Marshal(agent.Config())
+	if err != nil {
+		return fmt.Errorf("could not marshal Prow configuration: %v", err)
+	}
+
+	return ioutil.WriteFile(configPath, data, 0644)
```

Comment:
Why do we need to update the existing file?

Reply:
That's the whole point of the tool?

Reply:
Sry, I had a different meaning of `determinize` in my mind :)